### PR TITLE
Release notes for v4.0.0-preview3 

### PR DESCRIPTION
**Pull request by johnnypham**
Final version number TBD.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,33 @@ All notable changes to this project will be documented in this file.
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Preview Release 4.0.0-preview3.x.x] - 2021-10-20
+
+This update brings the below changes over the previous release:
+
+### Breaking changes over preview release v4.0.0-preview2
+
+- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)
+- Exception will not be thrown if a User ID is provided in the connection string when using `ActiveDirectoryIntegrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:14]: Could you mention this would be a breaking change on net fx?

File: release-notes/4.0/4.0.0-preview3.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+### Changed
+
+- Removed attributes for classes used in Microsoft.VSDesigner due to lack of support for Microsoft.Data.SqlClient [#1296](https://github.com/dotnet/SqlClient/pull/1296)
+- Disable encryption when connecting to SQL LocalDB [#1312](https://github.com/dotnet/SqlClient/pull/1312)
+
+## New features over preview release v4.0.0-preview2
+
+### Remove configurable retry logic safety switch
+
+The App Context switch "Switch.Microsoft.Data.SqlClient.EnableRetryLogic" will no longer be required to use the configurable retry logic feature. The feature is now supported in production. The default behavior of the feature will continue to be a non-retry policy, which will need to be overridden by client applications to enable retries.
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0-preview3.md:34]: This's been mentioned in Preview 2.

File: release-notes/4.0/4.0.0-preview3.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+This update brings the below changes over the previous release:
+
+### Breaking changes over preview release v4.0.0-preview2
+
+- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)
+- Exception will not be thrown if a User ID is provided in the connection string when using `ActiveDirectoryIntegrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
+
+### Added
+
+- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>`  support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019)
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0-preview3.md:14]: Could you refer to the [related topic below](https://github.com/dotnet/SqlClient/pull/1363/files#diff-148a966e6ab5b4bdd75f79a5fe4b4f743917096acf3ce84f9427a26792e5e76cR36) with a "read more" tag?

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,33 @@ All notable changes to this project will be documented in this file.
...
+
+### Added
+
+- Add `GetFieldValue(Async)<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019)
+
+### Fixed
+
+- Fixed `FormatException` when opening a connection with event tracing enabled [#1290](https://github.com/dotnet/SqlClient/pull/1290)
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:22]:
```suggestion
- Fixed `FormatException` when opening a connection with event tracing enabled [#1291](https://github.com/dotnet/SqlClient/pull/1291)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,33 @@ All notable changes to this project will be documented in this file.
...
+- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1327](https://github.com/dotnet/SqlClient/pull/1327)
+- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1335](https://github.com/dotnet/SqlClient/pull/1335)
+- Fixed data corruption issues by reverting changes to async cancellations [#1344](https://github.com/dotnet/SqlClient/pull/1344)
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:25]:
```suggestion
- Fixed data corruption issues by reverting changes to async cancellations [#1352](https://github.com/dotnet/SqlClient/pull/1352)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
...
+- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019)
+
+### Fixed
+
+- Fixed `FormatException` when opening a connection with event tracing enabled [#1291](https://github.com/dotnet/SqlClient/pull/1291)
+- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1327](https://github.com/dotnet/SqlClient/pull/1327)
+- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1335](https://github.com/dotnet/SqlClient/pull/1335)
+- Fixed data corruption issues by reverting changes to async cancellations [#1352](https://github.com/dotnet/SqlClient/pull/1352)
+- Fixed performance degradation by reverting changes to MARS state machine [#1357](https://github.com/dotnet/SqlClient/pull/1357)
+- Fixed bug where environment variables are ignored when using `ActiveDirectoryDefault` authentication [#1360](https://github.com/dotnet/SqlClient/pull/1360)
```
Comment (cheenamalhotra) [inline CHANGELOG.md:27]:
```suggestion
- Fixed bug where environment variables are ignored when using `Active Directory Default` authentication [#1360](https://github.com/dotnet/SqlClient/pull/1360)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Preview Release 4.0.0-preview3.x.x] - 2021-10-20
+
+This update brings the below changes over the previous release:
+
+### Breaking changes over preview release v4.0.0-preview2
+
+- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `ActiveDirectoryIntegrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
```
Comment (cheenamalhotra) [inline CHANGELOG.md:14]:
```suggestion
- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
+# Release Notes
+
+## Microsoft.Data.SqlClient 4.0.0-preview3.x.x released 20 October 2021
+
+This update brings the below changes over the previous release:
+
+### Breaking changes over preview release v4.0.0-preview2
+
+- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `ActiveDirectoryIntegrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
```
Comment (cheenamalhotra) [inline release-notes/4.0/4.0.0-preview3.md:10]:
```suggestion
- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019). [Read more](#`GetFieldValueAsync<T>`-and-`GetFieldValue<T>`-support-for-`XmlReader`,-`TextReader`,-`Stream`-types)
+
+### Fixed
+
+- Fixed `FormatException` when opening a connection with event tracing enabled [#1290](https://github.com/dotnet/SqlClient/pull/1290)
+- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1327](https://github.com/dotnet/SqlClient/pull/1327)
+- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1335](https://github.com/dotnet/SqlClient/pull/1335)
+- Fixed data corruption issues by reverting changes to async cancellations [#1344](https://github.com/dotnet/SqlClient/pull/1344)
+- Fixed performance degradation by reverting changes to MARS state machine [#1357](https://github.com/dotnet/SqlClient/pull/1357)
+- Fixed bug where environment variables are ignored when using `ActiveDirectoryDefault` authentication [#1360](https://github.com/dotnet/SqlClient/pull/1360)
```
Comment (cheenamalhotra) [inline release-notes/4.0/4.0.0-preview3.md:23]:
```suggestion
- Fixed bug where environment variables are ignored when using `Active Directory Default` authentication [#1360](https://github.com/dotnet/SqlClient/pull/1360)
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        connection.Open();
+        using (SqlDataReader reader = await command.ExecuteReaderAsync())
+        {
+            if (await reader.ReadAsync())
+            {
+                using (Stream stream = await reader.GetFieldValueAsync<Stream>(1))
+                {
+
+                }
```
Comment (cheenamalhotra) [inline release-notes/4.0/4.0.0-preview3.md:52]:
```suggestion
                {
                    // Continue to read from stream
                }
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+This update brings the below changes over the previous release:
+
+### Breaking changes over preview release v4.0.0-preview2
+
+- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
+
+### Added
+
+- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019). [Read more](#`GetFieldValueAsync<T>`-and-`GetFieldValue<T>`-support-for-`XmlReader`,-`TextReader`,-`Stream`-types)
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0-preview3.md:14]: It should be worked in preview mode too.
```suggestion
- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019). [Read more](#getfieldvalueasynct-and-getfieldvaluet-support-for-xmlreader-textreader-stream-types)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
...
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
+
+### Added
+
+- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019)
+
+### Fixed
+
+- Fixed `FormatException` when opening a connection with event tracing enabled [#1291](https://github.com/dotnet/SqlClient/pull/1291)
+- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1327](https://github.com/dotnet/SqlClient/pull/1327)
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:23]:
```suggestion
- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1328](https://github.com/dotnet/SqlClient/pull/1328)
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
+
+### Added
+
+- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019). [Read more](#getfieldvalueasynct-and-getfieldvaluet-support-for-xmlreader-textreader-stream-types)
+
+### Fixed
+
+- Fixed `FormatException` when opening a connection with event tracing enabled [#1290](https://github.com/dotnet/SqlClient/pull/1290)
+- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1327](https://github.com/dotnet/SqlClient/pull/1327)
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0-preview3.md:19]:
```suggestion
- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1328](https://github.com/dotnet/SqlClient/pull/1328)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -4,6 +4,34 @@ All notable changes to this project will be documented in this file.
...
+- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1335](https://github.com/dotnet/SqlClient/pull/1335)
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:24]:
```suggestion
- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1336](https://github.com/dotnet/SqlClient/pull/13365)
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1335](https://github.com/dotnet/SqlClient/pull/1335)
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0-preview3.md:20]:
```suggestion
- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1336](https://github.com/dotnet/SqlClient/pull/1336)
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+- Fixed data corruption issues by reverting changes to async cancellations [#1344](https://github.com/dotnet/SqlClient/pull/1344)
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0-preview3.md:21]:
```suggestion
- Fixed data corruption issues by reverting changes to async cancellations [#1352](https://github.com/dotnet/SqlClient/pull/1352)
```

File: release-notes/4.0/4.0.0-preview3.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
+
+### Added
+
+- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019). [Read more](#getfieldvalueasynct-and-getfieldvaluet-support-for-xmlreader-textreader-stream-types)
+
+### Fixed
+
+- Fixed `FormatException` when opening a connection with event tracing enabled [#1290](https://github.com/dotnet/SqlClient/pull/1290)
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0-preview3.md:18]:
```suggestion
- Fixed `FormatException` when opening a connection with event tracing enabled [#1291](https://github.com/dotnet/SqlClient/pull/1291)
```

**Review by DavoudEshtehari — APPROVED** [review]

**Pull request merged**